src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
         }
         Entry<K, V> evicted = head;
         head = evicted.getNextEntry();
-        head.setPreEntry(null);
+        if (head != null) {
+            head.setPreEntry(null);
+        }
```

Comment:
**Optimization Issue**: Missing null check before setting the previous entry to null can lead to NullPointerException. <br> **Fix**: Add a null check before setting the previous entry to ensure the head is not null. <br> **Code Suggestion**: 
 ```
 +        if (head != null) {
 +            head.setPreEntry(null);
 +        }
 ```